crates/assets/src/v1.rs

```diff
@@ -0,0 +1,213 @@
+		Self::ChainId(err)
+	}
+}
+
+impl From<AssetIdError> for AssetDidError {
+	fn from(err: AssetIdError) -> Self {
+		Self::AssetId(err)
+	}
+}
+
+/// An Asset DID as specified in the Asset DID method specification.
+#[derive(Clone, Eq, PartialEq, Ord, PartialOrd, RuntimeDebug, Encode, Decode, MaxEncodedLen, TypeInfo)]
+pub struct AssetDid {
+	pub chain_id: ChainId,
+	pub asset_id: AssetId,
+}
+
+impl AssetDid {
+	pub fn ether_currency() -> Self {
+		Self {
+			chain_id: Eip155Reference::ethereum_mainnet().into(),
+			asset_id: Slip44Reference(60.into()).into(),
+		}
+	}
+
+	pub fn bitcoin_currency() -> Self {
+		Self {
+			chain_id: ChainId::Bip122(GenesisHexHash32Reference::bitcoin_mainnet()),
+			asset_id: Slip44Reference(0.into()).into(),
+		}
+	}
+
+	pub fn litecoin_currency() -> Self {
+		Self {
+			chain_id: ChainId::Bip122(GenesisHexHash32Reference::litecoin_mainnet()),
+			asset_id: Slip44Reference(2.into()).into(),
+		}
+	}
+
+	pub fn dai_currency() -> Self {
+		Self {
+			chain_id: Eip155Reference::ethereum_mainnet().into(),
+			asset_id: EvmSmartContractFungibleReference(hex!("6b175474e89094c44da98b954eedeac495271d0f")).into(),
+		}
+	}
+
+	pub fn req_currency() -> Self {
+		Self {
+			chain_id: Eip155Reference::ethereum_mainnet().into(),
+			asset_id: EvmSmartContractFungibleReference(hex!("8f8221afbb33998d8584a2b05749ba73c37a938a")).into(),
+		}
+	}
+
+	pub fn cryptokitties_collection() -> Self {
+		Self {
+			chain_id: Eip155Reference::ethereum_mainnet().into(),
+			asset_id: AssetId::Erc721(EvmSmartContractNonFungibleReference(
+				EvmSmartContractFungibleReference(hex!("06012c8cf97BEaD5deAe237070F9587f8E7A266d")),
+				None,
+			)),
+		}
+	}
+
+	pub fn themanymatts_collection() -> Self {
+		Self {
+			chain_id: Eip155Reference::ethereum_mainnet().into(),
+			asset_id: AssetId::Erc1155(EvmSmartContractNonFungibleReference(
+				EvmSmartContractFungibleReference(hex!("28959Cf125ccB051E70711D0924a62FB28EAF186")),
+				None,
+			)),
+		}
+	}
+}
+
+impl AssetDid {
+	/// Try to parse an `AssetDID` instance from the provided UTF8-encoded
+	/// input.
+	pub fn from_utf8_encoded<I>(input: I) -> Result<Self, AssetDidError>
+	where
+		I: AsRef<[u8]> + Into<Vec<u8>>,
+	{
+		let input = input.as_ref();
+		let input_length = input.len();
+		if !(MINIMUM_ASSET_DID_LENGTH..=MAXIMUM_ASSET_DID_LENGTH).contains(&input_length) {
+			return Err(AssetDidError::InvalidFormat);
+		}
+
+		let asset_id = input
+			.as_ref()
+			.strip_prefix(DID_ASSET_PREFIX)
+			.ok_or(AssetDidError::InvalidFormat)?;
+
+		let mut split = asset_id.splitn(2, |c| *c == CHAIN_ASSET_SEPARATOR);
+		let (chain, asset) = (split.next(), split.next());
+
+		match (chain, asset) {
+			(Some(chain), Some(asset)) => {
+				let chain_id = ChainId::from_utf8_encoded(chain).map_err(AssetDidError::ChainId)?;
+				let asset_id = AssetId::from_utf8_encoded(asset).map_err(AssetDidError::AssetId)?;
+				Ok(Self { chain_id, asset_id })
+			}
+			_ => Err(AssetDidError::InvalidFormat),
+		}
```

Comment:
S: In case you are not planning on adding more match cases, it is recommended to use a simple `if else` block instead.

Reply:
I would honestly keep it like this, as I think is nicer (there are enum matches basically everywhere), and also easier to extend IF we add new components to an asset DID.